AST DeclRefExpr node has optional `non_odr_use_unevaluated` token; as used by `sizeof(x)`

The `DeclRefExpr` node has an optional `non_odr_use_unevaluated` token in Clang 9.0.

# Example

Contents of `j.c`:
```c
char a[123];

int x(void) {
	return sizeof(a);
}

```

# Panic

```
$ c2go transpile j.c
panic: could not match regexp with string
^(?P<address>[0-9a-fx]+) <(?P<position>.*)> '(?P<type>.*?)'(:'(?P<type1>.*?)')?.*?(?P<lvalue> lvalue)? (?P<for>\w+) (?P<address2>[0-9a-fx]+) '(?P<name>.*?)' '(?P<type2>.*?)'(:'(?P<type3>.*?)')?[\s]*$
0x55c856bd4970 <col:16> 'char [123]' lvalue Var 0x55c856bd4758 'a' 'char [123]' non_odr_use_unevaluated


goroutine 52 [running]:
github.com/elliotchance/c2go/ast.groupsFromRegex(0xc00032e0d0, 0xc7, 0xc000188639, 0x67, 0x7f0c7713d560)
	/home/u/goget/src/github.com/elliotchance/c2go/ast/ast.go:290 +0x360
github.com/elliotchance/c2go/ast.parseDeclRefExpr(0xc000188639, 0x67, 0x6b398b)

```

# AST output

The AST of `j.c` looks as follows:

```
$ c2go ast j.c
TranslationUnitDecl 0x55c4793959f8 <<invalid sloc>> <invalid sloc>
|-TypedefDecl 0x55c479396290 <<invalid sloc>> <invalid sloc> implicit __int128_t '__int128'
| `-BuiltinType 0x55c479395f90 '__int128'
|-TypedefDecl 0x55c479396300 <<invalid sloc>> <invalid sloc> implicit __uint128_t 'unsigned __int128'
| `-BuiltinType 0x55c479395fb0 'unsigned __int128'
|-TypedefDecl 0x55c4793965e8 <<invalid sloc>> <invalid sloc> implicit __NSConstantString 'struct __NSConstantString_tag'
| `-RecordType 0x55c4793963e0 'struct __NSConstantString_tag'
|   `-Record 0x55c479396358 '__NSConstantString_tag'
|-TypedefDecl 0x55c479396680 <<invalid sloc>> <invalid sloc> implicit __builtin_ms_va_list 'char *'
| `-PointerType 0x55c479396640 'char *'
|   `-BuiltinType 0x55c479395a90 'char'
|-TypedefDecl 0x55c479396958 <<invalid sloc>> <invalid sloc> implicit __builtin_va_list 'struct __va_list_tag [1]'
| `-ConstantArrayType 0x55c479396900 'struct __va_list_tag [1]' 1 
|   `-RecordType 0x55c479396760 'struct __va_list_tag'
|     `-Record 0x55c4793966d8 '__va_list_tag'
|-VarDecl 0x55c4793f4758 </home/u/Desktop/ccc/j/j.c:1:1, col:11> col:6 referenced a 'char [123]'
`-FunctionDecl 0x55c4793f48d0 <line:3:1, line:5:1> line:3:5 x 'int (void)'
  `-CompoundStmt 0x55c4793f49f8 <col:13, line:5:1>
    `-ReturnStmt 0x55c4793f49e8 <line:4:2, col:17>
      `-ImplicitCastExpr 0x55c4793f49d0 <col:9, col:17> 'int' <IntegralCast>
        `-UnaryExprOrTypeTraitExpr 0x55c4793f49b0 <col:9, col:17> 'unsigned long' sizeof
          `-ParenExpr 0x55c4793f4990 <col:15, col:17> 'char [123]' lvalue
            `-DeclRefExpr 0x55c4793f4970 <col:16> 'char [123]' lvalue Var 0x55c4793f4758 'a' 'char [123]' non_odr_use_unevaluated

```

--- a/ast/decl_ref_expr.go
+++ b/ast/decl_ref_expr.go
@@ -2,17 +2,18 @@ package ast
 
 // DeclRefExpr is expression.
 type DeclRefExpr struct {
-	Addr       Address
-	Pos        Position
-	Type       string
-	Type1      string
-	Lvalue     bool
-	For        string
-	Address2   string
-	Name       string
-	Type2      string
-	Type3      string
-	ChildNodes []Node
+	Addr                 Address
+	Pos                  Position
+	Type                 string
+	Type1                string
+	Lvalue               bool
+	For                  string
+	Address2             string
+	Name                 string
+	Type2                string
+	Type3                string
+	NonODRUseUnevaluated bool
+	ChildNodes           []Node
 }
 
 func parseDeclRefExpr(line string) *DeclRefExpr {
@@ -25,22 +26,24 @@ func parseDeclRefExpr(line string) *DeclRefExpr {
 		 (?P<address2>[0-9a-fx]+)
 		 '(?P<name>.*?)'
 		 '(?P<type2>.*?)'(:'(?P<type3>.*?)')?
+		(?P<non_odr_use_unevaluated> non_odr_use_unevaluated)?
 		`,
 		line,
 	)
 
 	return &DeclRefExpr{
-		Addr:       ParseAddress(groups["address"]),
-		Pos:        NewPositionFromString(groups["position"]),
-		Type:       groups["type"],
-		Type1:      groups["type1"],
-		Lvalue:     len(groups["lvalue"]) > 0,
-		For:        groups["for"],
-		Address2:   groups["address2"],
-		Name:       groups["name"],
-		Type2:      groups["type2"],
-		Type3:      groups["type3"],
-		ChildNodes: []Node{},
+		Addr:                 ParseAddress(groups["address"]),
+		Pos:                  NewPositionFromString(groups["position"]),
+		Type:                 groups["type"],
+		Type1:                groups["type1"],
+		Lvalue:               len(groups["lvalue"]) > 0,
+		For:                  groups["for"],
+		Address2:             groups["address2"],
+		Name:                 groups["name"],
+		Type2:                groups["type2"],
+		Type3:                groups["type3"],
+		NonODRUseUnevaluated: len(groups["non_odr_use_unevaluated"]) > 0,
+		ChildNodes:           []Node{},
 	}
 }
 

--- a/ast/decl_ref_expr_test.go
+++ b/ast/decl_ref_expr_test.go
@@ -71,6 +71,20 @@ func TestDeclRefExpr(t *testing.T) {
 			Type3:      "T_ENUM",
 			ChildNodes: []Node{},
 		},
+		`0x55e2113b7970 <col:16> 'char [123]' lvalue Var 0x55e2113b7758 'a' 'char [123]' non_odr_use_unevaluated`: &DeclRefExpr{
+			Addr:                 0x55e2113b7970,
+			Pos:                  NewPositionFromString("col:16"),
+			Type:                 "char [123]",
+			Type1:                "",
+			Lvalue:               true,
+			For:                  "Var",
+			Address2:             "0x55e2113b7758",
+			Name:                 "a",
+			Type2:                "char [123]",
+			Type3:                "",
+			NonODRUseUnevaluated: true,
+			ChildNodes:           []Node{},
+		},
 	}
 
 	runNodeTests(t, nodes)

